### PR TITLE
Ft 292 Draggable

### LINKED ISSUE #292: [Refactor]: Make draggable component
### Refactor Request

Currently AvatarDOM has the draggable property as well. draggable part and avatar part shoud be separate, so that it is possible to reuse AvatarDOM

~@CaseRegan I think this is worth an issue / PR before you finish your whiteboarding branch.
Would you be willing to make a PR just to make Draggable component?
I personally think doing this with functional component would be nice for consistency~

- https://github.com/Amijakan/mochatto/tree/case/whiteboarding

### PR DESCRIPTION
**Pull request by Njima1572**
## Which Issue was this PR made for?

- Closes #292 

## What is within the scope with this PR

- [x] Create Draggable Component
- [x] Detach AvatarDOM from position

## Other sidenotes that reviewer should be aware of
Position is written with jsx style prop, any idea on getting rid of it?

### CODE REVIEW

**Comment by Njima1572** [comment]
AUTOMATED: Deployed to https://pr-302.dev.mochatto.com

**Comment by Njima1572** [comment]
AUTOMATED: Deployed to https://pr-302.dev.mochatto.com

File: client/src/components/Draggable/index.tsx
Comment (siberowl) [inline client/src/components/Draggable/index.tsx:38]: I personally prefer the `updatePosition && updatePosition(newPosition)` syntax but doesn't really matter

File: client/src/components/Draggable/index.tsx
Comment (siberowl) [inline client/src/components/Draggable/index.tsx:7]: It's probably more intuitive and correct to name the callback function `onPositionChange`

**Review by siberowl — COMMENTED** [review]
There might be some repetitive things that can be removed in the AvatarDOM style sheet

**Comment by Njima1572** [comment]
AUTOMATED: Deployed to https://pr-302.dev.mochatto.com

**Comment by Njima1572** [comment]
AUTOMATED: Deployed to https://pr-302.dev.mochatto.com

**Review by siberowl — APPROVED** [review]
LGTM!
Let's wait until #221 gets merged to avoid conflicts there though

**Comment by Njima1572** [comment]
AUTOMATED: Deployed to https://pr-302.dev.mochatto.com

**Pull request merged**
